facet-core/src/impls_core/tuple.rs

```diff
@@ -32,2490 +29,214 @@ pub fn write_type_name_list(
+
+                                impl_facet_for_tuple! {
+                                    debug on f {
+                                        $(
+                                            unsafe {
+                                                let ptr = &value.$idx as *const $elems;
+                                                ($elems::SHAPE.vtable.debug.unwrap_unchecked())(
+                                                    PtrConst::new(ptr),
+                                                    f,
+                                                )
+                                            }?;
+                                        )+
+                                    }
+                                }
+                            });
+                        }
+
+                        if Characteristic::Default.all(elem_shapes) {
+                            builder = builder.default_in_place(|dst| {
+                                $(
+                                    unsafe {
+                                        ($elems::SHAPE.vtable.default_in_place.unwrap_unchecked())(
+                                            dst.field_uninit_at(mem::offset_of!(Self, $idx))
+                                        );
+                                    }
+                                )+
+
+                                unsafe { dst.assume_init() }
+                            });
+                        }
+
+                        if Characteristic::Clone.all(elem_shapes) {
+                             builder = builder.clone_into(|src, dst| {
+                                $({
+                                    let offset = mem::offset_of!(Self, $idx);
+                                    unsafe {
+                                        ($elems::SHAPE.vtable.clone_into.unwrap_unchecked())(
+                                            src.field(offset),
+                                            dst.field_uninit_at(offset),
+                                        );
+                                    }
+                                })+
+
+                                unsafe { dst.assume_init() }
+                            });
+                       }
+
+                        if Characteristic::PartialEq.all(elem_shapes) {
+                            builder = builder.eq(|a, b| impl_facet_for_tuple! {
+                                ord on ($($elems.$idx,)+),
+                                eq(a, b),
+                                eq = true
+                            });
+                        }
+
+                        if Characteristic::PartialOrd.all(elem_shapes) {
+                            builder = builder.partial_ord(|a, b| impl_facet_for_tuple! {
+                                ord on ($($elems.$idx,)+),
+                                partial_ord(a, b),
+                                eq = Some(Ordering::Equal)
+                            });
```

Comment:
these are so clean 😭

please add yourself to the facet-core authors (@veykril you too) in follow-up PRs